src/App.tsx

```diff
@@ -1,73 +1,71 @@
-import React from 'react';
+import React, { useMemo, useState } from 'react';
 import './App.scss';
+import classNames from 'classnames';
+
 import { peopleFromServer } from './data/people';
+import { Person } from './types/Person';
+
+import { SearchInput } from './components/searchInput/searchInput';
+import Content from './components/content/content';
 
 export const App: React.FC = () => {
-  const { name, born, died } = peopleFromServer[0];
+  const [delay, setDelay] = useState(300); // for change add setDelay
+
+  const [fullField, setFullField] = useState('');
+  const [selectedPerson, setSelectedPerson] = useState<Person | null>(null);
+
+  const [focus, setFocus] = useState(false);
+
+  const people = useMemo(() => {
+    return [...peopleFromServer];
+  }, []);
+
+  const peopleSearcher = useMemo(() => {
+    return people.filter(person => person.name.includes(fullField));
+  }, [people, fullField]);
```

Comment:
The filtering logic should be optimized to prevent re-renders when the input text hasn't changed. Consider adding a condition to check if `fullField` is different from the previous value before updating the `peopleSearcher`.